File: pythonProject/singleton.py
class Configuracao:
    _instancia = None

    def __new__(cls):
        if cls._instancia is None:
            cls._instancia = super().__new__(cls)
        return cls._instancia

# Testando o Singleton
config1 = Configuracao()
config2 = Configuracao()
print(config1 == config2)  # True, pois é a mesma instância

File: GoLangProject/singleton.go
package main

import (
	"fmt"
	"sync"
)

type Configuracao struct{}

var instancia *Configuracao
var once sync.Once

func GetInstancia() *Configuracao {
	once.Do(func() {
		instancia = &Configuracao{}
	})
	return instancia
}

func main() {
	config1 := GetInstancia()
	config2 := GetInstancia()
	fmt.Println(config1 == config2)
}
